Set default github.token

laughedelic opened the pull request:
We can't set it to `secrets.GITHUB_TOKEN` because `secrets` context is not in scope, but `github` context is, so this input doesn't have to be required and can use the provided `github.token` (same as `secrets.GITHUB_TOKEN`) by default.

Note: this doesn't break compatibility for those who set this input explicitly. Also, I didn't touch the deprecated `github_token`.

laughedelic committed:
Set default github.token

laughedelic committed:
Update README.md

laughedelic committed:
don't set token in the CI workflow

mikepenz commented:
@laughedelic thank you so much for the PR. I'll have a look over it in the next few days. 

May you elaborate though please why `secrets.GITHUB_TOKEN` is not in scope for your case? I was in the assumption that it is a default provided secret

laughedelic commented:
> May you elaborate though please why `secrets.GITHUB_TOKEN` is not in scope for your case? I was in the assumption that it is a default provided secret

Hi! The problem is not that I can't use the secret in my workflow, I meant that secrets are not in scope of the _action definition_. So we can't say 

```yaml
inputs:
  token:
    description: 'Specify the token to use to publish the check.'
    required: false
    default: ${{ secrets.GITHUB_TOKEN }}
```

It will error, saying that `secrets` cannot be accessed here. But the same default value _can_ be set with `github.token` instead. It's exactly the same token, just a different way to refer to it.

This change is not for any particular use case and doesn't solve any problems, it just makes it a bit easier to use it, e.g.

```yaml
- uses: mikepenz/action-junit-report@v2
  with:
    report_paths: '**/target/test-reports/*.xml'
```

no need to set the `token` or `github_token` inputs.

laughedelic committed:
Change workflow trigger to pull_request_target

https://github.blog/2020-08-03-github-actions-improvements-for-fork-and-pull-request-workflows/#improvements-for-public-repository-forks

laughedelic commented:
I noticed that it failed to create the checks, but I think it's because of the fork. I changed the trigger to `pull_request_target` ([details](https://github.blog/2020-08-03-github-actions-improvements-for-fork-and-pull-request-workflows/#improvements-for-public-repository-forks)), I hope it helps.

mikepenz commented:
@laughedelic thank you so much for elaborating. I see now what you mean, and that makes indeed sense. 

The last commit (before changing the trigger) looks to have had passed the checks: https://github.com/mikepenz/action-junit-report/runs/2445076612
(had manually approved the PR to be executed as first time contributor)

laughedelic commented:
@mikepenz it passed, but the checks haven't been published. If you open the [summary](https://github.com/mikepenz/action-junit-report/actions/runs/786571211) of that run you will see messages like

> Failed to create checks using the provided token. (HttpError: Resource not accessible by integration)

I pushed one more commit: a456feb, but it didn't trigger the workflow run. I'm not sure what is the problem with it 🤔 

P.S. I also see that happening on [other PRs](https://github.com/mikepenz/action-junit-report/actions/runs/775113545).

mikepenz commented:
Super strange. no other messages or errors shown for me either. would be great if github had more messaging for such cases. in any way will review this in the next couple of days and get it in. 

Thank you so much!

mikepenz commented:
Thank you so much once again for the PR
